ta4j-core/src/test/java/org/ta4j/core/analysis/criteria/NumberOfTradesCriterionTest.java

```diff
@@ -53,21 +53,22 @@ public void calculateWithTwoTrades() {
                 Order.buyAt(3, series), Order.sellAt(5, series));
 
         AnalysisCriterion buyAndHold = new NumberOfTradesCriterion();
-        assertEquals(2d, buyAndHold.calculate(series, tradingRecord), TestUtils.GENERAL_OFFSET);
+        assertNumEquals(2, buyAndHold.calculate(series, tradingRecord));
     }
 
     @Test
     public void calculateWithOneTrade() {
+        MockTimeSeries series = new MockTimeSeries(numFunction, 100, 105, 110, 100, 95, 105);
         Trade trade = new Trade();
         NumberOfTradesCriterion tradesCriterion = new NumberOfTradesCriterion();
 
-        assertEquals(1d, tradesCriterion.calculate(null, trade), TestUtils.GENERAL_OFFSET);
+        assertNumEquals(1, tradesCriterion.calculate(series, trade));
     }
 
     @Test
     public void betterThan() {
         AnalysisCriterion criterion = new NumberOfTradesCriterion();
```

Comment:
- [ ] Please change `new NumberOfTradesCriterion()` to `getCriterion()`.

Reply:
It's not possible as there is no factory defined in this test (I get a , also none of the other Criterions do it that way, am I missing something?

This is the exception I get if I change it:

```java
java.lang.NullPointerException
	at org.ta4j.core.analysis.criteria.AbstractCriterionTest.getCriterion(AbstractCriterionTest.java:71)
	at org.ta4j.core.analysis.criteria.NumberOfBarsCriterionTest.betterThan(NumberOfBarsCriterionTest.java:69)
[..]
```java